src/main/java/com/cst438/service/RegistrarServiceProxy.java

```diff
@@ -0,0 +1,305 @@
+    private void addSection(String s) throws Exception {
+        SectionDTO sectionDTO = fromJsonString(s, SectionDTO.class);
+
+        Course course = courseRepository.findById(sectionDTO.courseId()).orElse(null);
+        if (course == null ){
+            throw new Exception("Course not found: " + sectionDTO.courseId());
+        }
+
+        Section section = new Section();
+        section.setCourse(course);
+
+        Term term = termRepository.findByYearAndSemester(sectionDTO.year(), sectionDTO.semester());
+        if (term == null) {
+            throw new Exception("year, semester invalid");
+        }
+        section.setTerm(term);
+        section.setSecId(sectionDTO.secId());
+        section.setBuilding(sectionDTO.building());
+        section.setRoom(sectionDTO.room());
+        section.setTimes(sectionDTO.times());
+
+        User instructor = null;
+        if (sectionDTO.instructorEmail()==null || sectionDTO.instructorEmail().equals("")) {
+            section.setInstructor_email("");
+        } else {
+            instructor = userRepository.findByEmail(sectionDTO.instructorEmail());
+            if (instructor == null || !instructor.getType().equals("INSTRUCTOR")) {
+                throw new Exception("email not found or not an instructor " + sectionDTO.instructorEmail());
+            }
+            section.setInstructor_email(sectionDTO.instructorEmail());
+        }
+        sectionRepository.save(section);
+    }
+    private void updateSection(String s) throws Exception {
+        SectionDTO sectionDTO = fromJsonString(s, SectionDTO.class);
+        Section section = sectionRepository.findById(sectionDTO.secNo()).orElse(null);
+
+        if (s==null) {
+            throw new Exception("section not found "+sectionDTO.secNo());
+        }
+        section.setSecId(sectionDTO.secId());
+        section.setBuilding(sectionDTO.building());
+        section.setRoom(sectionDTO.room());
+        section.setTimes(sectionDTO.times());
+
+        User instructor = null;
+        if (sectionDTO.instructorEmail()==null || sectionDTO.instructorEmail().equals("")) {
+            section.setInstructor_email("");
+        } else {
+            instructor = userRepository.findByEmail(sectionDTO.instructorEmail());
+            if (instructor == null || !instructor.getType().equals("INSTRUCTOR")) {
+                throw new Exception("email not found or not an instructor " + sectionDTO.instructorEmail());
+            }
+            section.setInstructor_email(sectionDTO.instructorEmail());
+        }
+        sectionRepository.save(section);
+    }
+    private void deleteSection(String s) throws Exception {
+        SectionDTO dto = fromJsonString(s, SectionDTO.class);
+        Section sec = sectionRepository.findById(dto.secNo()).orElse(null);
+        if (sec != null) {
+            throw new Exception("Section contains assignments or enrollmens: " + dto.secNo());
+        }
+        sectionRepository.deleteById(dto.secNo());
```

Comment:
Delete operations just take an primary key (id), not an entire DTO. And this one is an int so we will have to parse from a string. So something like this:
```
        try {
            Integer id = Integer.parseInt(s);
            sectionRepository.deleteById(id);
        } catch (NumberFormatException e) {
            throw e;
        }